mono/io-layer/wait.c

```diff
@@ -68,7 +69,7 @@ guint32 WaitForSingleObjectEx(gpointer handle, guint32 timeout,
 	guint32 ret, waited;
 	int thr_ret;
 	gboolean apc_pending = FALSE;
-	gpointer current_thread = wapi_get_current_thread_handle ();
+	gpointer current_thread = mono_thread_info_get_handle (mono_thread_info_current ());
 	gint64 now, end;
 	
 	if (current_thread == NULL) {
```

Comment:
isn't this condition always false now?  you assert in mono_thread_info_get_handle that info->handle isn't null.